src/backend/dwarf.c

```diff
@@ -790,6 +790,20 @@ void writeEhFrameFDE(IDXSEC dfseg, Symbol *sfunc)
     Outbuffer *buf = SegData[dfseg]->SDbuf;
     const unsigned startsize = buf->size();
 
+    if (sfunc->ty() & mTYnaked)
+    {
+        /* Do not have info on naked functions. Assume they are set up as:
+         *   push RBP
+         *   mov  RSP,RSP
+         */
+        int off = 2 * REGSIZE;
+        dwarf_CFA_set_loc(1);
+        dwarf_CFA_set_reg_offset(SP, off);
+        dwarf_CFA_offset(BP, -off);
+        dwarf_CFA_set_loc(I64 ? 4 : 3);
+        dwarf_CFA_set_reg_offset(BP, off);
+    }
+
```

Comment:
The existing stack unwinding mechanism already assumes a normal stack frame. This does not make things worse, it brings it up to the current state by emitting eh_frame info that says it's a normal stack frame.

We can do better by extending the inline assembler to support CFA pseudo-ops, but that would break existing code, and is a non-trivial project itself.

The stack frame info is necessary only if the inline assembler does not call any functions that throw. The etc.linux.memoryerror does do this. I considered removing memoryerror, as it converts seg faults into thrown exceptions, something that doesn't belong in a systems programming language, but decided to simply make it work for now and leave that issue for a separate day.